BlueMap/Chat/src/main/java/codes/antti/auth/bluemap/chat/BlueMapChatPlugin.java

```diff
@@ -46,12 +49,39 @@ public void forEachSession(SessionOperator op) {
 	@Override
 	public void onLoad() {
 		BlueMapAPI.onEnable(api -> {
+			this.config = getConfig();
+			var plugin_manager = getServer().getPluginManager();
+			boolean bmjs_moved = false;
+
+			// Copy edited js file
 			try {
+				var intConfig = Objects.requireNonNull(plugin_manager.getPlugin("BlueMap-Auth"));
+				this.integrationConfig = intConfig.getConfig();
+				String integration = new String(Objects.requireNonNull(getResource("bluemap-chat.js")).readAllBytes(), StandardCharsets.UTF_8)
+						.replaceAll("\\{\\{auth-path}}", Objects.requireNonNull(this.integrationConfig.getString("auth-path", "/authentication-outpost/")))
+						.replaceAll("\\{\\{web-chat-prefix}}", Objects.requireNonNull(this.config.getString("web-chat-prefix", "[web]")))
+						.replaceAll("\\{\\{max-message-count}}", Objects.requireNonNull(this.config.getString("max-message-count", "100")));
+				Path chatPath = api.getWebApp().getWebRoot().resolve("assets/bluemap-chat.js");
+				Files.createDirectories(chatPath.getParent());
+				OutputStream out = Files.newOutputStream(chatPath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+				out.write(integration.getBytes(StandardCharsets.UTF_8));
+				out.close();
 				api.getWebApp().registerScript("assets/bluemap-chat.js");
+				bmjs_moved = true;
+			}
+			catch (Exception ex) {
+				getLogger().severe("Could not load bluemap-chat.js");
+				ex.printStackTrace();
+			}
+
+			try {
 				api.getWebApp().registerStyle("assets/bluemap-chat.css");
-				copyResource("bluemap-chat.js");
+
 				copyResource("bluemap-chat.css");
 				copyResource("minecraft.otf");
+				if (!bmjs_moved) {
+					copyResource("bluemap-chat.js");
+				}
```

Comment:
In my opinion this bmjs_moved stuff is useless, if it failed it failed, don't see the need to retry without the placeholders being filled.

Reply:
One of the placeholders are from the Authentication plugin. To keep the plugin working, I had a fallback.

Reply:
I guess it becomes unnecessary when the presence of bluemap-auth has been made optional